test/tools/ossfuzz/SolidityGenerator.h

```diff
@@ -152,7 +152,11 @@ class TestCaseGenerator: public GeneratorBase
 		return "Test case generator";
 	}
 private:
-	std::string path() const
+	/// Returns a new source path name that is formed by concatenating
+	/// a static prefix @name m_sourceUnitNamePrefix, a monotonically
+	/// increasing counter starting from 0 and the postfix (extension)
+	/// ".sol".
+	[[nodiscard]] std::string path() const
```

Comment:
Why is `[[nodiscard]]` important here?

Reply:
iiuc, if I were to discard the output of `path()` in some expression e.g., `path();`  instead of `string p = path();` AND the function `path()` contains the annotation `[[nodiscard]]`, then the compiler warns on discarding-of-no-discard-value

```
error: ignoring return value of function declared with 'nodiscard' attribute [-Werror,-Wunused-result]
                path();
                ^~~~
```

Without this annotation, the warning goes away.

Reply:
Yeah, but why is it important that the value returned by `path` is not discarded?

Reply:
I also don't understand the point of using `[[nodiscard]]`.